umh.docs.umh.app/content/en/docs/features/connectivity/_index.md

```diff
@@ -1,6 +1,128 @@
+
+The Connection Management functionality in the Management Console aims to
+address those challenges by providing a simple and easy to use tool to manage
+all the assets in your factory.
+
+You can add, delete, and most importantly, visualize the status of all your
+connections in a single place. For example, a periodic check is performed to
+measure the latency of each connection, and the status of the connection is
+displayed in the Management Console.
+
+You can also add notes to each connection, so that you can keep all the documentation
+in a single place.
+
+![Connection Notes](/images/features/connectivity/management-console-connection-management-notes.png?)
+
+You can then configure a data source for each connection, to start extracting
+data from your assets. Once the data source is configured, specific information
+about its status is displayed, prompting you in case of misconfigurations, data
+not being received, or other any error that may occur.
+
+## How can I use it?
+
+Add new connections from the Connection Management page of the Management Console.
+Then, configure a data source for each of them by choosing one of the available
+tools, depending on the type of connection.
+
+The following tools come with the United Manufacturing Hub and are recommended
+for extracting data from your assets:
+
+### Node-RED
+
+[Node-RED](https://nodered.org/) is a leading open-source tool for IIoT
+connectivity. We recommend this tool for prototyping and integrating parts of the
+shop floor that demand high levels of customization and domain knowledge.
+
+Even though it may be unreliable in high-throughput scenarios, it has a vast
+global community that provides a wide range of connectors for different protocols
+and data sources, while remaining very user-friendly with its visual programming
+approach.
+
+### Benthos UMH
+
+<!-- TODO: write a page in the Architecture section about Benthos, and consider also a reference page -->
+
+Benthos UMH is a custom extension of the [Benthos](https://www.benthos.dev/)
+project. It allows you to connect assets that communicate via the OPC UA protocol,
+and it is recommended for scenarios involving the extraction of large data volumes
+in a standardized format.
+
+It is a lightweight, open-source tool that is easy to deploy and manage. It is
+ideal for moving medium-sized data volumes more reliably then Node-RED, but it
+requires some technical knowledge.
+
+### Other Tools
+
+The United Manufacturing Hub also provides tools for connecting data sources
+that uses other types of connections. For example, you can easily connect
+[ifm IO-Link](/docs/features/connectivity/additionalconnectivity/ifm-retrofitting/)
+sensors or [USB barcode readers](/docs/features/connectivity/additionalconnectivity/barcodereader-retrofitting/).
+
+### Third-Party Tools
+
+Virtually any existing connectivity solution can be integrated with the United
```

Comment:
Any existing connectivity solution can be integrated with the United Manufacturing Hub, assuming it can send data to either MQTT or Kafka. Additionally, if you want to deploy those tools on the Device & Container Infrastructure, they must be available as a Docker container (developed with best-practices).

Why the last part with best-practices? you can also provide a docker container, where you need to mount the full host system and which accesses libraries there. Or they store data not in volumes, but instead hidden somewhere in the file system. Or are not configurable via a text file, but instead include hteir own database, etc.
Is absolutely stupid, but I've seen this too often.

Reply:
updated